### ISSUE TITLE
Implement Rate Limiting for API Abuse Prevention and Fair Usage

### ISSUE THREAD

**Issue by momanyisamuel** [activist-org/activist#362]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This is an issue to implement rate limiting. Rate limiting helps prevent abuse, ensure fair usage, and protect the system from excessive requests that can lead to performance degradation or denial of service.

### Contribution

Happy to support someone on this issue or work on it myself

**Comment by kurtisolo** [comment]
Could it be something like this? What would be some 'sensible' values?
```
REST_FRAMEWORK = {
    'DEFAULT_THROTTLE_CLASSES': [
        'rest_framework.throttling.AnonRateThrottle',
        'rest_framework.throttling.UserRateThrottle'
    ],
    'DEFAULT_THROTTLE_RATES': {
        'anon': 'XXXX/hour',
        'user': 'XXXX/hour'
    }
}
```

**Comment by sulabhkatila** [comment]
Hi, I'd love to work on this. I'd appreciate some guidance on getting started with the issue. 
Thank you!

**Comment by andrewtavis** [comment]
Hey @sulabhkatila 👋 Thanks for your interest in activist and in helping out! I'd be fine with you sending along the changes after a discussion of what should be done 😊 The documentation for this can be found [here](https://www.django-rest-framework.org/api-guide/throttling/#userratethrottle). I'd say for this issue it'd even be enough to get this into the code so that we can quickly react to it if there's a rate limit problem. I'm open for opinions on what the actual rates for `anon` and `user` should be, especially given that we won't have users for still some time.

@sulabhkatila, do you want to do some research and report back, and then we can discuss where the changes should happen?

**Comment by andrewtavis** [comment]
Hey @LmOpe! Thanks for your interest in helping out :) There hasn't been traction here for a month, so I'd be happy to assign this to you and we can figure this out. We could do a call, or you'd be welcome to join us in our developer sync that's happening in an hour (details are in the Development channel in our [Matrix community](https://matrix.to/#/#activist_community:matrix.org) - super last minute, so no stess!). Aside from this, maybe @momanyisamuel or @kurtisolo would want to provide some direction in the meantime :)

Thanks again for your interest!

**Comment by LmOpe** [comment]
Hello @andrewtavis, I would love to jump on a call with you for proper discussion if that's fine.

**Comment by sulabhkatila** [comment]
Hi @LmOpe @andrewtavis, I made a [PR](https://github.com/activist-org/activist/pull/421). 
Please be aware that there might be issues with the testing part. I encountered errors while trying to run the tests myself, and I wasn't able to resolve them. Feel free to comment on it and even improve on it.

**Comment by andrewtavis** [comment]
Hey @sulabhkatila 👋 Thanks for the PR, and sorry for assuming no traction :) Happy to discuss this issue with you further in the PR as is needed 😊

@LmOpe, I'll keep you in mind for new issues that are made, and please keep an eye out for new [`good first issues`](https://github.com/activist-org/activist/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)! :)

**Comment by sulabhkatila** [comment]
Hi @andrewtavis, I'm eager to make the necessary improvements and complete this feature with your guidance.

**Comment by andrewtavis** [comment]
Closed via #421 :) :) Thank you, @sulabhkatila! Looking forward to working with you again, and sorry the review took a bit 🙂

@to-sta, can you make the testing issue for this?

**Comment by to-sta** [comment]
> @to-sta, can you make the testing issue for this?

Just to check track #432. Will add an issue as soon as I finish the setup 😃 for testing
